src/Features/LanguageServer/ProtocolUnitTests/Ordering/RequestOrderingTests.cs

```diff
@@ -120,6 +121,32 @@ public async Task ThrowingTaskDoesntBringDownQueue()
             Assert.All(responses, r => Assert.True(r.EndTime > r.StartTime));
         }
 
+        [Fact]
+        public async Task LongRunningSynchronousNonMutatingTaskDoesNotBlockQueue()
+        {
+            var requests = new[] {
+                new TestRequest(LongRunningNonMutatingRequestHandler.MethodName),
+                new TestRequest(MutatingRequestHandler.MethodName),
+                new TestRequest(NonMutatingRequestHandler.MethodName),
+            };
+
+            using var testLspServer = CreateTestLspServer("class C { }", out _);
+
+            // Cancel all requests if we hang for 5 seconds. This will result in a failed test run.
+            using var cts = new CancellationTokenSource(TimeSpan.FromSeconds(5));
```

Comment:
The test seems to rely on everything else taking less than 5 seconds, which should usually be true, but seems potentially flaky (for example, a random 5 second gc pause while running).  I wonder if we could make this deterministic by having two handlers - the first request being a long running handler which doesn't finish _until_ a flag is set by the handler handling the second request.  That way we don't have to rely on a specific time span for the test to run (though we may want to cancel with the hang mitigating timeout at some point).

Reply:
Extended to 1 minute